client/my-sites/stats/activity-log-tasklist/plugins-to-update.jsx

```diff
@@ -0,0 +1,56 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { unionBy } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getPluginsWithUpdates } from 'state/plugins/installed/selectors';
+
+export default WrappedComponent => {
+	class PluginsToUpdate extends Component {
+		static propTypes = {
+			siteId: PropTypes.number,
+			pluginsWithUpdates: PropTypes.arrayOf( PropTypes.object ),
+		};
+
+		state = {
+			// Plugins with pending updates
+			pluginsWithUpdates: [],
+			// Plugins already updated + those with pending updates
+			allPluginsToUpdate: [],
+		};
+
+		componentWillReceiveProps( nextProps ) {
+			if ( this.state.pluginsWithUpdates.length < nextProps.pluginsWithUpdates.length ) {
```

Comment:
watch out! this logic is testing an indirect measure and will fail when we get a _different_ set of the _same length_ as the previous set of plugins

Reply:
Good catch. Thanks. WIll fix.